Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp

```diff
@@ -1050,5 +1085,233 @@ namespace AzToolsFramework
+            {
+                SavePrefabsInDialog(prefabSaveSelectionDialog.get());
+            }
+            return prefabSaveSelection;
+        }
+
+        void PrefabIntegrationManager::ExecuteSavePrefabDialog(TemplateId templateId, bool useSaveAllPrefabsPreference)
+        {
+            using namespace AzToolsFramework::Prefab;
+
+            auto prefabTemplate = s_prefabSystemComponentInterface->FindTemplate(templateId);
+            AZ::IO::Path prefabTemplatePath = prefabTemplate->get().GetFilePath();
+
+            if (s_prefabSystemComponentInterface->IsTemplateDirty(templateId))
+            {
+                if (s_prefabLoaderInterface->SaveTemplate(templateId) == false)
+                {
+                    AZ_Error("Prefab", false, "Template '%s' could not be saved successfully.", prefabTemplatePath.c_str());
+                    return;
+                }
+            }
+
+            if (useSaveAllPrefabsPreference)
+            {
+                SaveAllPrefabsPreference saveAllPrefabsPreference = s_prefabLoaderInterface->GetSaveAllPrefabsPreference();
+
+                if (saveAllPrefabsPreference == SaveAllPrefabsPreference::SaveAll)
+                {
+                    s_prefabSystemComponentInterface->SaveAllDirtyTemplates(templateId);
+                    return;
+                }
+                else if (saveAllPrefabsPreference == SaveAllPrefabsPreference::SaveNone)
+                {
+                    return;
+                }
+            }
+
+            AZStd::unique_ptr<QDialog> savePrefabDialog = ConstructSavePrefabDialog(templateId, useSaveAllPrefabsPreference);
+            if (savePrefabDialog)
+            {
+                int prefabSaveSelection = savePrefabDialog->exec();
+
+                if (prefabSaveSelection == QDialog::Accepted)
+                {
+                    SavePrefabsInDialog(savePrefabDialog.get());
+                }
+            }
+        }
+
+        void PrefabIntegrationManager::SavePrefabsInDialog(QDialog* unsavedPrefabsDialog)
+        {
+            QList<QLabel*> unsavedPrefabFileLabels = unsavedPrefabsDialog->findChildren<QLabel*>(UnsavedPrefabFileName);
+            if (unsavedPrefabFileLabels.size() > 0)
+            {
+                for (const QLabel* unsavedPrefabFileLabel : unsavedPrefabFileLabels)
+                {
+                    AZStd::string unsavedPrefabFileName = unsavedPrefabFileLabel->property("FilePath").toString().toUtf8().data();
+                    AzToolsFramework::Prefab::TemplateId unsavedPrefabTemplateId =
+                        s_prefabSystemComponentInterface->GetTemplateIdFromFilePath(unsavedPrefabFileName.data());
+                    bool isTemplateSavedSuccessfully = s_prefabLoaderInterface->SaveTemplate(unsavedPrefabTemplateId);
+                    AZ_Assert(isTemplateSavedSuccessfully, "Prefab '%s' could not be saved successfully.", unsavedPrefabFileName.c_str());
```

Comment:
I think this should be a AZ_Error instead, as it is typed to the data

Reply:
At this point, a user would have asked their unsaved templates to be saved to files. If a template is loaded correctely but couldn't be saved correctly, then it is an error in our systems. A user won't be able to easily fix it. So, an assert makes more sense here. Thoughts?

Reply:
Keep in mind that a user that doesn't run with VS attached is going to get their editor crashed at this point, hence why it may be preferable to be an error as its not a programmer error, but data or IO error

Reply:
Changed to AZ_Error to let the user save as many prefabs as possible without crashing on the first failure.